### PR TITLE
feat: spawn the wait/update for session in comms:is_sent

### PR DESCRIPTION
**Pull request by joshuef**

### CODE REVIEW

File: sn_node/src/comm/mod.rs [hunk 1]
```diff
@@ -156,55 +155,62 @@ impl Comm {
...
-                    Err(error) => match error {
-                        Error::PeerLinkDropped => {
-                            // remove the peer link
-                            let perhaps_session = self.sessions.remove(&peer);
-                            if let Some((_peer, session)) = perhaps_session {
-                                session.disconnect().await;
+        let sessions = self.sessions.clone();
+
+        // TODO: we could cache the handles above and check them as part of loop...
+        let _handle = tokio::spawn(async move {
```
Comment (maqi) [inline sn_node/src/comm/mod.rs:161]: just wondering, as there is no calling of `_handle.await` to allow a waiting on this spawned task, hence the returns on the error branches, such as `return Err(Error::PeerLinkDropped);` or `Err(Error::FailedSend(peer))` actually has no effect on reporting back the errors during send?

File: sn_node/src/comm/peer_session.rs [hunk 2]
```diff
@@ -43,8 +43,7 @@ impl PeerSession {
     pub(crate) fn new(link: Link) -> PeerSession {
         let (sender, receiver) = mpsc::channel(1000);
 
-        let _ =
-            tokio::task::spawn_local(PeerSessionWorker::new(link, sender.clone()).run(receiver));
+        let _ = tokio::task::spawn(PeerSessionWorker::new(link, sender.clone()).run(receiver));
```
Comment (maqi) [inline sn_node/src/comm/peer_session.rs:46]: just wondering, shall we check all the places using `spawn_local` to confirm it's really being used properly?
I feel that for the most cases, we shall use `spawn` as default ?

File: sn_node/src/comm/mod.rs (hunk 1 above)
Comment (joshuef) [reply sn_node/src/comm/mod.rs:161]: Yup, this is the loop we need to close but without blocking in the send channel. Right now we can't send any other message until we have something to report. Instead I think we need a channel to feed back reports to the node perhaps?

**Comment by joshuef** [comment]
Closing as #1614 and #1615 are both ahead of this

**Pull request closed without merging**